### ISSUE TITLE
Pandas v1.3.0 [BREAKING CHANGE]: Pandas indexing changed -> expected list, got FrozenList

### ISSUE THREAD

**Issue by G-kodes** [jnothman/UpSetPlot#141]
I am unable to run the package at all. At first, I thought it was my data, but even when I changed my generation methods to those indicated in the docs, I still got the same error message.

I am running a Mambaforge implementation if it means anything (Packages all installed fine?). From what I can tell, the upsetplot function `plot_matrix()` in `plotting.py` (Link Below) attempts to use pandas column/indices selection (i.e. `data[column-name]`) which is not designed to take a `FrozenList` instance, which is what `data.index.names` returns?

https://github.com/jnothman/UpSetPlot/blob/master/upsetplot/plotting.py#L536

## What I ran:
```python
from upsetplot import plot, generate_counts

plot(generate_counts())
```

## The Error I received:
```bash
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
d:\GitHub\Pharmacogenetics-Snakemake\workflow\scripts\04 - GRAPHS.py in <module>
----> 1 plot(generate_counts())

D:\mambaforge\envs\snakemake\lib\site-packages\upsetplot\plotting.py in plot(data, fig, **kwargs)
    715         Keys are 'matrix', 'intersections', 'totals', 'shading'
    716     """
--> 717     return UpSet(data, **kwargs).plot(fig)

D:\mambaforge\envs\snakemake\lib\site-packages\upsetplot\plotting.py in plot(self, fig)
    669         matrix_ax = self._reorient(fig.add_subplot)(specs['matrix'],
    670                                                     sharey=shading_ax)
--> 671         self.plot_matrix(matrix_ax)
    672         totals_ax = self._reorient(fig.add_subplot)(specs['totals'],
    673                                                     sharey=matrix_ax)

D:\mambaforge\envs\snakemake\lib\site-packages\upsetplot\plotting.py in plot_matrix(self, ax)
    502         n_cats = data.index.nlevels
    503 
--> 504         idx = np.flatnonzero(data.index.to_frame()[data.index.names].values)
    505         c = np.array(['lightgrey'] * len(data) * n_cats, dtype='O')
    506         c[idx] = self._facecolor

D:\mambaforge\envs\snakemake\lib\site-packages\pandas\core\frame.py in __getitem__(self, key)
   3443 
   3444         # Do we have a (boolean) 1d indexer?
-> 3445         if com.is_bool_indexer(key):
   3446             return self._getitem_bool_array(key)
   3447 

D:\mambaforge\envs\snakemake\lib\site-packages\pandas\core\common.py in is_bool_indexer(key)
    144     elif isinstance(key, list):
    145         # check if np.array(key).dtype would be bool
--> 146         return len(key) > 0 and lib.is_bool_list(key)
    147 
    148     return False

TypeError: Argument 'obj' has incorrect type (expected list, got FrozenList)
```

**Comment by G-kodes** [comment]
Further investigation has revealed there is a slight change in indexing methods between Pandas `v1.2.0` and `v1.3.0` as shown here:
https://github.com/pandas-dev/pandas/issues/42461

this would explain why this method no-longer works

**Comment by W-L** [comment]
Thanks for figuring this out! I ran into the same issue on a fresh conda install and can confirm that downgrading `pandas` to 1.2.4 solves the problem.
`conda install pandas=1.2.4`

**Comment by jnothman** [comment]
Thanks for the report. I'm inclined to not release a fix since this is to be fixed in Pandas 1.3.1 (https://github.com/pandas-dev/pandas/pull/42469). Downgrade pandas for now.

**Comment by jnothman** [comment]
Release looks imminent: https://github.com/pandas-dev/pandas/issues/42343
